src/outputs/parthenon_hdf5.cpp

```diff
@@ -372,14 +461,99 @@ void PHDF5Output::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) {
       WRITEH5SLAB("z", tmpData.data(), gLocations, local_start, local_count, global_count,
                   property_list);
     }
+    {
+      // open locations tab
+      H5G const gVLocations = H5G::FromHIDCheck(
+          H5Gcreate(file, "/VolumeLocations", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+
+      // write X coordinates
+      local_count[0] = num_blocks_local;
+      global_count[0] = max_blocks_global;
+
+      // These macros are defined in parthenon_hdf5.hpp, which establishes relevant scope
+      LOADVARIABLEALL(tmpData, pm, pmb->coords.x1v, out_ib.s, out_ib.e, 0, 0, 0, 0);
+      local_count[1] = global_count[1] = nx1;
+      WRITEH5SLAB("x", tmpData.data(), gVLocations, local_start, local_count,
+                  global_count, property_list);
+
+      // write Y coordinates
+      LOADVARIABLEALL(tmpData, pm, pmb->coords.x2v, 0, 0, out_jb.s, out_jb.e, 0, 0);
+      local_count[1] = global_count[1] = nx2;
+      WRITEH5SLAB("y", tmpData.data(), gVLocations, local_start, local_count,
+                  global_count, property_list);
+
+      // write Z coordinates
+      LOADVARIABLEALL(tmpData, pm, pmb->coords.x3v, 0, 0, 0, 0, out_kb.s, out_kb.e);
+
+      local_count[1] = global_count[1] = nx3;
+      WRITEH5SLAB("z", tmpData.data(), gVLocations, local_start, local_count,
+                  global_count, property_list);
+    }
+
+    {
+      // Write Levels and Logical Locations with the level for each Meshblock
+      // loclist contains levels and logical locations for all meshblocks on
+      // all ranks
+      std::vector<LogicalLocation> loclist = pm->GetLocList();
```

Comment:
Can  you make this a const reference, doesn't look like you are altering it's contents.